test/mongoose_graphql_SUITE.erl

```diff
@@ -870,73 +881,165 @@ use_dir_module_and_service_not_loaded(Config) ->
+use_dir_module_service_and_db_loaded(Config) ->
+    Doc = <<"{catA { command4(domain: \"localhost\")} }">>,
+    Ctx = #{},
+    {Ast, Ctx2} = check_directives(Config, Ctx, Doc),
+    Res = execute_ast(Config, Ctx2, Ast),
+    ?assertEqual(#{data => #{<<"catA">> => #{<<"command4">> => <<"command4">>}}}, Res).
+
+use_dir_db_not_loaded(Config) ->
+    Doc = <<"{catA { command5(domain: \"localhost\")} }">>,
+    Ctx = #{},
+    {Ast, Ctx2} = check_directives(Config, Ctx, Doc),
+    #{errors := [Error]} = execute_ast(Config, Ctx2, Ast),
+    #{extensions :=
+        #{code := deps_not_loaded,
+          not_loaded_internal_databases := [<<"db_x">>]},
+      message :=
+        <<"Some of the required internal databases are not loaded">>,
+      path := [<<"catA">>, <<"command5">>]
+     } = Error.
+
+use_dir_module_service_and_db_not_loaded(Config) ->
+    Doc = <<"{catA { command6(domain: \"localhost\")} }">>,
+    Ctx = #{},
+    {Ast, Ctx2} = check_directives(Config, Ctx, Doc),
+    #{errors := [Error]} = execute_ast(Config, Ctx2, Ast),
+    #{extensions :=
+        #{code := deps_not_loaded,
+          not_loaded_modules := [<<"mod_x">>],
+          not_loaded_services := [<<"service_x">>],
+          not_loaded_internal_databases := [<<"db_x">>]},
+      message :=
+        <<"Some of the required modules and services and internal databases are not loaded">>,
+      path := [<<"catA">>, <<"command6">>]
+     } = Error.
+
+use_dir_object_all_modules_services_and_dbs_loaded(Config) ->
     Doc = <<"{ catC { command(domain: \"localhost\") } }">>,
     Ctx = #{},
     {Ast, Ctx2} = check_directives(Config, Ctx, Doc),
     Res = execute_ast(Config, Ctx2, Ast),
     ?assertEqual(#{data => #{<<"catC">> => #{<<"command">> => <<"command">>}}}, Res).
 
-use_dir_object_module_and_service_not_loaded(Config) ->
+use_dir_object_module_and_db_not_loaded(Config) ->
+    Doc = <<"{ catD { command(domain: \"localhost\") } }">>,
+    Ctx = #{},
+    {Ast, Ctx2} = check_directives(Config, Ctx, Doc),
+    #{errors := [Error]} = execute_ast(Config, Ctx2, Ast),
+    #{extensions :=
+        #{code := deps_not_loaded,
+          not_loaded_modules := [<<"mod_x">>],
+          not_loaded_internal_databases := [<<"db_x">>]},
+    message :=
+        <<"Some of the required modules and internal databases are not loaded">>,
+      path := [<<"catD">>, <<"command">>]
+     } = Error.
+
+use_dir_object_service_and_db_not_loaded(Config) ->
+    Doc = <<"{ catD { command3 } }">>,
+    Ctx = #{user => jid:make_bare(<<"user">>, <<"localhost">>)},
+    {Ast, Ctx2} = check_directives(Config, Ctx, Doc),
+    #{errors := [Error]} = execute_ast(Config, Ctx2, Ast),
+    #{extensions :=
+        #{code := deps_not_loaded,
+          not_loaded_services := [<<"service_x">>],
+          not_loaded_internal_databases := [<<"db_x">>]},
+          message := <<"Some of the required services and internal databases are not loaded">>,
+      path := [<<"catD">>, <<"command3">>]
```

Comment:
```suggestion
          path := [<<"catD">>, <<"command3">>]
```

Reply:
I think the `message` element is incorrectly indented here. It should be on the same nesting level as `path` and `extensions`.